trunk/SUAVE/Components/Energy/Converters/Rotor.py

```diff
@@ -0,0 +1,781 @@
+            
+            dR_dpsi[np.isnan(dR_dpsi)] = 0.1
+                      
+            dpsi   = -Rsquiggly/dR_dpsi
+            psi    = psi + dpsi
+            diff   = np.max(abs(psiold-psi))
+            psiold = psi
+            
+            # If its really not going to converge
+            if np.any(psi>(pi*85.0/180.)) and np.any(dpsi>0.0):
+                broke = True
+                break
+                
+            ii+=1
+                
+            if ii>2000:
+                broke = True
+                break
+            
+        # There is also RE scaling
+        #This is an atrocious fit of DAE51 data at RE=50k for Cd
+        Cdval = (0.108*(Cl*Cl*Cl*Cl)-0.2612*(Cl*Cl*Cl)+0.181*(Cl*Cl)-0.0139*Cl+0.0278)*((50000./Re)**0.2)
+        Cdval[alpha>=pi/2] = 2.
+        
+        # More Cd scaling from Mach from AA241ab notes for turbulent skin friction
+        Tw_Tinf = 1. + 1.78*(Ma*Ma)
+        Tp_Tinf = 1. + 0.035*(Ma*Ma) + 0.45*(Tw_Tinf-1.)
+        Tp      = (Tp_Tinf)*T
+        Rp_Rinf = (Tp_Tinf**2.5)*(Tp+110.4)/(T+110.4)
+        
+        Cd = ((1/Tp_Tinf)*(1/Rp_Rinf)**0.2)*Cdval  
+        epsilon  = Cd/Cl
+        epsilon[epsilon==np.inf] = 10. 
+        deltar   = (r[1]-r[0])
+        thrust   = rho*B*(np.sum(Gamma*(Wt-epsilon*Wa)*deltar,axis=1)[:,None])
+        torque   = rho*B*np.sum(Gamma*(Wa+epsilon*Wt)*r*deltar,axis=1)[:,None]
+        D        = 2*R 
+        tip_speed = omega*R
+ 
+        # Leishman's thrust coefficient for rotor
+        Ctl        = thrust/(rho*disk_area*( tip_speed*tip_speed))  # Eqn 2.36 Principles of Helicopter Aerodynamics 
+        Ctl[Ctl<0] = 0.        # prevent things from breaking
+        
+        # motor thrust coefficient
+        Ct       = thrust/(rho*(n*n)*(D*D*D*D)) # used for motor model
+        Ct[Ct<0] = 0.  
+        Cd0      = self.profile_drag_coefficient   
+        Cp       = np.zeros_like(Ct)
+        Cpl      = np.zeros_like(Ct)
+        power    = np.zeros_like(Ct) 
+        for i in range(len(Vv)): 
+            if -1. < Vv[i][0] < 1.: # vertical/axial flight 
+                Cpl[i]      = (kappa*(Ctl[i]**1.5)/(2**.5))+sigma*Cd0/8. # Eqn 2.43 Principles of Helicopter Aerodynamics 
+                power[i]    = Cpl[i]*rho[i]*disk_area*(tip_speed[i]*tip_speed[i]*tip_speed[i]) 
+                torque[i]   = power[i]/omega[i]    
+            else:   
+                power[i]    = torque[i]*omega[i]   
+                torque[i]   = power[i]/omega[i]  
+            Cp[i] = power[i]/(rho[i]*(n[i]*n[i]*n[i])*(D*D*D*D*D))  
+            
+        thrust[conditions.propulsion.throttle[:,0] <=0.0] = 0.0
+        power[conditions.propulsion.throttle[:,0]  <=0.0] = 0.0 
+        thrust[omega<0.0] = - thrust[omega<0.0] 
+
+        etap     = V*thrust/power     
+        
+        conditions.propulsion.etap = etap        
+        
+        # store data
+        results_conditions = Data     
+        outputs            = results_conditions(
```

Comment:
Stored data should have verbose variable names.

Reply:
variables explicitly defined